### PR TITLE
Added policy for Compute Engine disk resource policies.

### PR DESCRIPTION
**Pull request by rosmo**
Adding a policy to check for disk `resourcePolicies` (eg. snapshot schedules). Supports both `Disk` and `RegionDisk`. Also supports an empty policy ("").

### CODE REVIEW

File: policies/templates/gcp_compute_disk_resource_policies_v1.yaml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+# This template is for policies restricting the locations 
+# of compute (VM instance / persistent disk) resources to
+# specific zones in GCP. It supports allowlist or denylist
+# modes, as well as exempting selected assets from the list.
+
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-compute-disk-resource-policies-v1
+  annoations:
```
Comment (morgante) [inline policies/templates/gcp_compute_disk_resource_policies_v1.yaml]:
```suggestion
  annoations: {}
```

Or just leave annotations off entirely.

File: policies/templates/gcp_compute_disk_resource_policies_v1.yaml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+              description: "String identifying the operational mode, allowlist or denylist. In allowlist mode, 
+              disk resource policies are only allowed in the policies specified in the 'resource_policies' parameter. 
+              In denylist mode, all disk resources policies are allowed except those listed in the 'resource_policies' 
+              parameter."
+            exemptions:
+              type: array
+              items: string
+              description: "Array of disk assets to exempt from resource policies restriction. String values in the array 
+              should correspond to the full name values of exempted policies."
+            zones:
```
Comment (morgante) [inline policies/templates/gcp_compute_disk_resource_policies_v1.yaml]: Is this meant to be `resourcePolicies`?

File: validator/compute_disk_resource_policies_test.rego [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+# Test for no violations with no assets
+test_compute_disk_resource_policies_no_assets {
+	found_violations := find_violations with data.assets as []
+
+	count(found_violations) == 0
+}
+
+# Test for no violations with no constraints
```
Comment (morgante) [inline validator/compute_disk_resource_policies_test.rego]: Can you update tests to use the helpers from #289?

File: validator/test/fixtures/compute_disk_resource_policies/assets/data.json [hunk 4]
```diff
@@ -0,0 +1,147 @@
+[
```
Comment (morgante) [inline validator/test/fixtures/compute_disk_resource_policies/assets/data.json]: In the test fixtures, could we include another (non-disk) asset to verify [this check](https://github.com/forseti-security/policy-library/pull/297/files#diff-fbf18d3450e452021b8e6161186a6991R33) is correct?

**Comment by rosmo** [comment]
Sorry, I messed up my repository, will do a new PR.

**Pull request closed without merging**